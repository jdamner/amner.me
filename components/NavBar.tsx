import React from "react";
/* Components */
import { HomeIcon } from "@heroicons/react/24/solid";
import InlineLink from "./Atoms/Links/Inline";

export default function NavBar(
  props: React.DetailedHTMLProps<
    React.HTMLAttributes<HTMLElement>,
    HTMLElement
  >,
) {
  const { children, ...rest } = props;
  return (
    <nav
      className="w-full h-16 bg-gradient-to-r text-slate-100 from-slate-900 to-slate-700 dark:text-slate-200"
      id="main-nav"
      {...rest}
    >
      <div className="container mx-auto h-full flex flex-row items-center px-3 md:px-0">
        <div className="">
          <InlineLink href="/" className="text-slate-100 hover:text-slate-300">
            <HomeIcon className="h-10" />
            <span className="sr-only">Home</span>
          </InlineLink>
        </div>
        <div className="ml-auto">{children}</div>
      </div>
    </nav>
  );
}
