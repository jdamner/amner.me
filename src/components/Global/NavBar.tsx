import Link from "next/link"
import InlineLink from "../Links/Inline";

export default function NavBar(props): JSX.Element {
	return (
		<nav className="w-full h-16 bg-gradient-to-r bg-emerald-900 text-slate-100 from-teal-900 to-emerald-600">
			<div className='container mx-auto h-full flex flex-row justify-between items-center px-3 md:px-0'>
				<ul>
					<li className="inline-block mr-3">
						<InlineLink href="/" className="text-xl font-bold">amner.me</InlineLink>
					</li>
				</ul>
				{props.children}
			</div>
		</nav>
	);
}
