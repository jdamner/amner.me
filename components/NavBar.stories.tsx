import type { Meta, StoryObj } from "@storybook/nextjs";

import NavBar from "./NavBar";

type Story = StoryObj<typeof NavBar>;

export default {
  title: "Components/NavBar",
  component: NavBar,
} as Meta;

export const Default: Story = {
  args: {
    children: "Children",
  },
};
