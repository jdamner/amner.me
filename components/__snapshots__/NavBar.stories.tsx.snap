// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/NavBar Default smoke-test 1`] = `
<div class="bg-black text-orange-100  font-sans">
  <header class="w-full text-black bg-orange-600 p-4"
          id="main-nav"
  >
    <div class="container mx-auto px-3">
      <a class="font-black uppercase text-5xl"
         href="/"
         style="font-family: &quot;Bebas Neue&quot;; font-style: normal; font-weight: 400;"
      >
        James Amner
      </a>
    </div>
  </header>
</div>
`;
